### PR TITLE
Implement deployed infrastructure in terraform

### PR DESCRIPTION
**Pull request by RyanWhite25**
## Context

<-- Describe the context surrounding the change you're making -->

## Changes proposed in this pull request

<-- Describe the changed you've made, a bullet point list of each change is enough -->

## Guidance to review

<-- How can someone who is reviewing this see your changes or reproduce the fix -->

## Things to check

- [ ] Styling has not broken
- [ ] Content changes have been reviewed and approved
- [ ] No sensitive information about the team or our upcoming work has been included

### CODE REVIEW

**Comment by dk-singh** [comment]
Do we need to push `terraform/.terraform.lock.hcl` file

File: terraform/cloudfront.tf [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  aliases     = [for url in var.web_domain_hostnames : url]
+  web_acl_id  = aws_wafv2_web_acl.website.arn
+
+  is_ipv6_enabled = true
+
+  default_cache_behavior {
+    allowed_methods = ["GET", "HEAD"]
+    cached_methods  = ["GET", "HEAD"]
+
+    cache_policy_id  = "658327ea-f89d-4fab-a63d-7e88639e58f6" # CachingOptimized
```
Comment (dk-singh) [inline terraform/cloudfront.tf:35]: What is the hash value for `cache_policy_id` ?

Comment (RyanWhite25) [reply terraform/cloudfront.tf:35]: The guid is needed as this is what the CloudFront API expects to set the `CachingOptimized` managed caching policy. I've put a comment beside the value indicating the type of caching policy.

There's some further info on this here:
https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/using-managed-cache-policies.html#managed-cache-caching-optimized

**Comment by RyanWhite25** [comment]
> Do we need to push `terraform/.terraform.lock.hcl` file

Yes, including this is best practice as it ensures provider version upgrades are tracked in version control and prevents unexpected version upgrades between CI runs as newer provider versions are released. It's similar to a package-lock.json or poetry.lock

**Review by dk-singh — APPROVED** [review]
lgtm

**Pull request merged**